app/main.cpp

```diff
@@ -264,7 +264,10 @@ int main(int argc, char *argv[])
             }
         }
 
-        const QByteArray &featureArch = features.architecture();
+        auto featureArch = features.architecture();
```

Comment:
this sounds odd... I'm reluctant to merge this without understanding more about this, but I can see how it would help. I think we need to ask about this on the perf ML first, to figure out why it's not x86 in the first place. if the two arches are similar enough but different, I'd rather have the code deal with it both explicitly, instead of overriding the arch here. I.e. wherever we check for x86, also check for i586